keel-core/src/main/kotlin/com/netflix/spinnaker/keel/core/api/ArtifactSummary.kt

```diff
@@ -69,3 +70,23 @@ data class StatefulConstraintSummary(
   val comment: String? = null,
   val attributes: ConstraintStateAttributes? = null
 )
+
+@JsonInclude(Include.NON_NULL)
+data class StatelessConstraintSummary(
+  val type: String,
+  val currentlyPassing: Boolean,
+  val status: ConstraintStatus? = null,
+  val comment: String? = null,
+  val attributes: ConstraintMetadata? = null
+)
+
+abstract class ConstraintMetadata()
+
+data class DependOnConstraintMetadata(
+  val environment: String
+) : ConstraintMetadata()
+
+data class AllowedTimesConstraintMetadata(
+  val windows: List<TimeWindow>,
```

Comment:
does this preserve whatever string format existed on the persisted delivery config itself? i.e. if they put in `2-20` as the `hours` string will it be represented exactly like that?

Reply:
nevermind — just noticed your updated PR description has an example with a dashedd.

I'm not sure whether it's appropriate to do it in this PR or not, but for UI purposes it'd be better to split this up into a more structured object, e.g.:
```js
{
  "days": "Wed",
  "hours": {
    from: "12"
    to: "14"
  }
}
```

Reply:
yes, it does.. do you prefer it to be shown differently?

Reply:
I'm gonna open a GH issue for this as I don't think it's critical to getting things working — but I would like to make the attributes data for these much more structured before we fully move on from working on constraint APIs so it doesn't stay this way forever :D

Reply:
Yeah, I agree with you - I prefer it more structured as well. I can make the change in a separate pr.

Reply:
👍 thanks gal!